src/var.jl

```diff
@@ -114,19 +143,37 @@ function instantiate(::Type{NonCommutative{O}}) where {O}
     return NonCommutative(instantiate(O))
 end
 
+@enum ComplexKind cpNone cpFull cpConj cpReal cpImag
```

Comment:
Could you add a docstring here ?

Reply:
Done - should this then be exported?

Reply:
Yes, I like to follow https://jump.dev/JuMP.jl/stable/developers/style/#Exported-and-non-exported-names

Reply:
Can we rename the constants with capital letters ? What about : `REAL`, `COMPLEX`, `CONJ`, `REAL_PART`, `IMAG_PART` ? `CONJ` and `IMAG` could also be expanded but this is consistent with `Base.conj` and `Base.imag`

Reply:
Yes, it's ok to not export then.

> Documenter doesn't like non-exported docstrings.

What do you mean ? DynamicPolynomials doesn't have a documentation anyway.

> With the prefix (which unfortunately isn't so widespread for enums outside of Pascal-like languages) at least the danger of collisions is reduced.

I'm not a big fan of prefix like `cp` which reduces readability. I prefer non-exported long names like `DynamicPolynomials.REAL` than exported short names with a prefix to try to avoid collision. The way we deal with collision in the JuMP style guide is that we advise to use `import` but we don't have to follow this guide strictly so let's not export and have short, simple names without cryptic prefix

Reply:
Ok, renamed, but kept unexported.
> What do you mean ? 

If I remember correctly, Documenter warns (fails if not excepted) if you have a method with a docstring that is not exported. Perhaps it also was a method with a docstring that is not included in the documentation, I don't remember. Which kind of makes sense: A docstring is for user documentation, if you want developer documentation, use a comment (maybe).

Well, prefixes are good if there's no way to do scoping in the language. With the modularization in Julia, this is already better as explicit package prefixes are now required. Still, a clean naming of enums IMHO means that the name of the enum value makes clear where it belongs to. This is possible is the language offers/requires you to write `Enumtype.enumvalue`, but since we all like expressive names, `Enumtype` might not be very short, so this unnecessarily makes both writing and reading harder. Short expressive prefixes are a very good compromise that also help for code completion without the full might of prefixing by the enum name.